[2025-01-21T01:25:43.470+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-21T01:25:43.491+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: manage_airflow_roles.list_roles manual__2025-01-21T01:24:29.188645+00:00 [queued]>
[2025-01-21T01:25:43.503+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: manage_airflow_roles.list_roles manual__2025-01-21T01:24:29.188645+00:00 [queued]>
[2025-01-21T01:25:43.503+0000] {taskinstance.py:2866} INFO - Starting attempt 2 of 2
[2025-01-21T01:25:43.522+0000] {taskinstance.py:2889} INFO - Executing <Task(_PythonDecoratedOperator): list_roles> on 2025-01-21 01:24:29.188645+00:00
[2025-01-21T01:25:43.534+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=289) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-01-21T01:25:43.535+0000] {standard_task_runner.py:72} INFO - Started process 291 to run task
[2025-01-21T01:25:43.536+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'manage_***_roles', 'list_roles', 'manual__2025-01-21T01:24:29.188645+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpixburo6l']
[2025-01-21T01:25:43.538+0000] {standard_task_runner.py:105} INFO - Job 9: Subtask list_roles
[2025-01-21T01:25:43.836+0000] {task_command.py:467} INFO - Running <TaskInstance: manage_airflow_roles.list_roles manual__2025-01-21T01:24:29.188645+00:00 [running]> on host 8a59744600df
[2025-01-21T01:25:44.000+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data-platform' AIRFLOW_CTX_DAG_ID='manage_***_roles' AIRFLOW_CTX_TASK_ID='list_roles' AIRFLOW_CTX_EXECUTION_DATE='2025-01-21T01:24:29.188645+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-21T01:24:29.188645+00:00'
[2025-01-21T01:25:44.002+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-01-21T01:25:44.003+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-01-21T01:25:44.004+0000] {logging_mixin.py:190} INFO - Current task name:list_roles state:running start_date:2025-01-21 01:25:43.491865+00:00
[2025-01-21T01:25:44.004+0000] {logging_mixin.py:190} INFO - Dag name:manage_***_roles and current dag run status:running
[2025-01-21T01:25:44.005+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-21T01:25:49.265+0000] {logging_mixin.py:190} INFO - CompletedProcess(args=['***', 'roles', 'list'], returncode=0, stdout='name  \n======\nAdmin \nOp    \nPublic\nUser  \nViewer\n      \n', stderr='')
[2025-01-21T01:25:49.268+0000] {logging_mixin.py:190} INFO - aqui
[2025-01-21T01:25:49.269+0000] {logging_mixin.py:190} INFO - <class 'subprocess.CompletedProcess'>
[2025-01-21T01:25:49.269+0000] {python.py:240} INFO - Done. Returned value was: CompletedProcess(args=['airflow', 'roles', 'list'], returncode=0, stdout='name  \n======\nAdmin \nOp    \nPublic\nUser  \nViewer\n      \n', stderr='')
[2025-01-21T01:25:49.284+0000] {xcom.py:690} ERROR - Object of type CompletedProcess is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2025-01-21T01:25:49.287+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 92, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 190, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'subprocess.CompletedProcess'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 789, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session_or_null)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3639, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 249, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 688, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 105, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 94, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type CompletedProcess is not JSON serializable
[2025-01-21T01:25:49.314+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-01-21T01:25:49.315+0000] {logging_mixin.py:190} INFO - Task start:2025-01-21 01:25:43.491865+00:00 end:2025-01-21 01:25:49.314073+00:00 duration:5.822208
[2025-01-21T01:25:49.316+0000] {logging_mixin.py:190} INFO - Task:<Task(_PythonDecoratedOperator): list_roles> dag:<DAG: manage_***_roles> dagrun:<DagRun manage_***_roles @ 2025-01-21 01:24:29.188645+00:00: manual__2025-01-21T01:24:29.188645+00:00, state:running, queued_at: 2025-01-21 01:25:41.387902+00:00. externally triggered: True>
[2025-01-21T01:25:49.316+0000] {logging_mixin.py:190} INFO - Failure caused by Object of type CompletedProcess is not JSON serializable
[2025-01-21T01:25:49.317+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=manage_***_roles, task_id=list_roles, run_id=manual__2025-01-21T01:24:29.188645+00:00, execution_date=20250121T012429, start_date=20250121T012543, end_date=20250121T012549
[2025-01-21T01:25:49.355+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-21T01:25:49.357+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 9 for task list_roles (Object of type CompletedProcess is not JSON serializable; 291)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 92, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 190, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'subprocess.CompletedProcess'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 789, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session_or_null)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3639, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 249, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 688, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 105, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 94, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type CompletedProcess is not JSON serializable
[2025-01-21T01:25:49.404+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-21T01:25:49.426+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-21T01:25:49.428+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
